rasa/core/agent.py

```diff
@@ -295,14 +296,11 @@ def __init__(
     ):
         # Initializing variables with the passed parameters.
         self.domain = self._create_domain(domain)
-        if self.domain:
-            self.domain.add_requested_slot()
         self.policy_ensemble = self._create_ensemble(policies)
-        if not self._is_form_policy_present():
-            raise InvalidDomain(
-                "You have defined a form action, but haven't added the "
-                "FormPolicy to your policy ensemble."
-            )
+
+        if self.domain is not None:
+            self.domain.add_requested_slot()
+            PolicyEnsemble.check_missing_policies(self.policy_ensemble, self.domain)
```

Comment:
Should we not check this even if no domain is present?

Reply:
Why?

Reply:
But there can be policies requiring certain to be present in the domain

Reply:
Are there any like that?